packages/trader/src/App/Components/Elements/PositionsDrawer/positions-drawer.tsx

```diff
@@ -109,6 +137,7 @@ const PositionsDrawer = observer(({ ...props }) => {
         <DataList
             data_source={positions}
             rowRenderer={args => (
+                // @ts-expect-error not sure how to type row so types are compatible
```

Comment:
@henry-deriv  what kind of error do you get? I'd like to help :)

Reply:
`Type '{ row?: TRow | undefined; is_footer?: boolean | undefined; columns_map?: Record<TColIndex, { key?: string | undefined; title?: ReactNode; col_index: TColIndex; renderCellContent?: ((props: TRenderCellContent) => ReactNode) | undefined; renderHeader?: ((props: THeaderProps) => ReactNode) | undefined; } | undefined> |...' is not assignable to type '{ currency: string; getContractById: (contract_id?: number | undefined) => TContractStore; is_new_row?: boolean | undefined; measure?: (() => void) | undefined; onClickRemove: (id: number) => void; row: TPortfolioPosition; server_time: Moment | undefined; symbol: string; }'.
  Types of property 'row' are incompatible.
    Type 'TRow | undefined' is not assignable to type 'TPortfolioPosition'.
      Type 'undefined' is not assignable to type 'TPortfolioPosition'.ts(2322)
const PositionsDrawerCardItem: ({ row: portfolio_position, measure, onHoverPosition, symbol, is_new_row, ...props }: TPositionDrawerCardItem) => JSX.Element`

Essentially, `TRow` which is `[key: string]: any` is not compatible to TPortfolioPosition which comes from `types.ts`. Of course i can see that [key: string]: any basically applies to any object like TPortfolioPosition but typesscript doesnt like it.

Reply:
@henry-deriv I can see you didn't add any type for `PositionsDrawer` component **props**:
`const PositionsDrawer = observer(({ ...props }: TPositionsDrawerProps) => {`
in its props you can add a suitable type for row which most likely will be
`row?: { [key: string]: any } | TPortfolioPosition;`
:)

Reply:
yeah i did that before and got rid of the error. I just thought it was an ugly solution